spec/requests/all_casa_admins/casa_admins_spec.rb

```diff
@@ -19,50 +27,101 @@
         }.to change(CasaAdmin, :count).by(1)
       end
     end
+
+    context "with invalid parameters" do
+      it "renders new page" do
+        post all_casa_admins_casa_org_casa_admins_path(casa_org), params: {
+          casa_admin: {
+            email: "",
+            display_name: ""
+          }
+        }
+        expect(response).to be_successful
+        expect(response).to render_template "casa_admins/new"
+      end
+    end
   end
 
   describe "PATCH /update" do
     let(:email) { "casa_admin@example.com" }
-    let(:casa_admin) {
-      CasaAdmin.new(
-        email: "admin1@example.com",
-        display_name: "Example Admin"
-      )
-    }
-    let(:other_admin) {
-      CasaAdmin.new(
-        email: "other_admin@example.com",
-        display_name: "Other Admin"
-      )
-    }
-
-    context "when current user is all casa admin or casa org admin" do
+
+    context "when current user is all casa admin" do
       it "should allow current user to successfully update other casa admin's email" do
-        casa_admin.update(email: "new_email@example.com")
+        patch all_casa_admins_casa_org_casa_admin_path(casa_org, casa_admin),
+          params: {all_casa_admin: {email: email}}
+
+        expect(response).to redirect_to edit_all_casa_admins_casa_org_casa_admin_path(casa_org, casa_admin)
+        expect(flash[:notice]).to eq("New admin created successfully")
+        expect(casa_admin.reload.email).to eq(email)
+      end
 
-        expect(casa_admin.email).to eq("new_email@example.com")
+      it "should render edit page if update fails" do
+        patch all_casa_admins_casa_org_casa_admin_path(casa_org, casa_admin),
+          params: {all_casa_admin: {email: ""}}
+
+        expect(response).to be_successful
+        expect(response).to render_template "casa_admins/edit"
       end
+    end
+  end
 
+  describe "PATCH /deactivate" do
+    context "when current user is all casa admin" do
       it "should successfully deactivate another casa admin's profile" do
-        casa_admin.active = true
-        casa_admin.deactivate
```

Comment:
`casa_admin.deactivate` and `casa_admin.activate` are both already being tested in spec/models/casa_admin_spec.rb so I changed these lines to access the controller methods.